src/targets/gpu/kernels/include/migraphx/kernels/group_query_attention.hpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#ifndef MIGRAPHX_GUARD_KERNELS_GROUP_QUERY_ATTENTION_HPP
+#define MIGRAPHX_GUARD_KERNELS_GROUP_QUERY_ATTENTION_HPP
+
+#include <migraphx/kernels/index.hpp>
+#include <migraphx/kernels/algorithm.hpp>
+#include <migraphx/kernels/integral_constant.hpp>
+#include <migraphx/kernels/tensor_view.hpp>
+#include <limits>
+#include <migraphx/kernels/type_traits.hpp>
+
+namespace migraphx {
+struct gqa_parameters
+{
+    float scale;
+    int batch_size;           // Batch size used by input
+    int sequence_length;      // Sequence length used by input
+    int hidden_size;          // Hidden size used by input
+    int head_size;            // Head size
+    int rotary_embedding_dim; // Rotary embedding dimension.
+    int num_heads;            // num_heads = hidden_size / head_size
+    int max_sequence_length;  // Sequence length used by cos/sin cache
+    int head_stride;          // Head stride
+    int seq_stride;           // Sequence stride
+    int batch_stride;         // Batch stride
+    int position_ids_format;  // Format of position ids - 0 is (1), 1 is (batch_size,
+                              // sequence_length)
+    int transposed; // Whether the input tensor has been transposed into (batch, num_heads,
+                    // seq_len, hidden)
+    int seqlen_present_kv_cache;
+
+    int do_rotary;
+    int kv_num_heads;
+    int local_window_size;
+    int rotary_interleaved;
+    int past_present_share_buffer;
+
+    __host__ __device__ void print() const
```

Comment:
Use `operator<<` instead, similar to how the `shape` class is used:

```cpp
template <class Stream>
friend constexpr const Stream& operator<<(const Stream& ss, const gqa_parameters& gp)
{
    ss << "scale: " << scale << "\n";
    ...
    return ss;
}
```

This way we can print this using our print function.